### ISSUE TITLE
[Request] Rewards API

### ISSUE THREAD

**Issue by fasterthanlime** [itchio/itch.io#903]
Originally opened by @SkyTech6 at https://github.com/itchio/itch/issues/1194

---

It would be great if through the server-side API we could check to see if a player has purchased a Reward for our games. This way it can be used for in-game rewards such as kickstarter bonuses (or even DLC).

I was thinking something along the lines of 

GET https://itch.io/api/1/KEY/game/GAME_ID/REWARD_ID/download_keys

with the user_id, download_key, email parameters available as with the other API calls.

Of course this would also require that we could see "Reward IDs" in our Dashboard perhaps. Maybe even one of the Fields can be one where we can assign the reward an ID?

**Comment by SkyTech6** [comment]
Any news on this one @fasterthanlime ? <3

**Comment by fasterthanlime** [comment]
So, there is a `claimed-rewards` endpoint but only if you're using an API key linked to an account that has admin access for the game. Also there's no filtering by e-mail / user id, it just returns *all the rewards*.

I don't think this is really what you wanted right @SkyTech6 ?

**Comment by SkyTech6** [comment]
No, this would need to be a way to check if a specific user has purchased a reward. 

I was thinking an endpoint very similar to `https://itch.io/api/1/KEY/game/GAME_ID/download_keys` except for Rewards on a specific game. `https://itch.io/api/1/KEY/game/GAME_ID/REWARD_ID/download_keys`. 

That way if someone is using the Itch App (or other method to get Itch account api keys) then you can see if they've purchased specific Rewards to give them in-game bonuses. Essentially adding DLC options to Itch.io.

[Example Image](https://i.imgur.com/xvc3FYD.png)

**Comment by SkyTech6** [comment]
@fasterthanlime I noticed the api documentation now has a "sub-product"? Is this able to be checked with /download_keys in some way?

**Comment by leafo** [comment]
@SkyTech6 sub products are a different feature from rewards. They are designed for things more similar to DLC. This is the only public API we have for them right now: https://itch.io/docs/api/sub-products

Sub-products are currently in limited release and only available to approved accounts. Tell me if you're interested in using them.

**Comment by llemoi** [comment]
So I can see there's the claimed_rewards endpoint, but unless I'm mistaken theres's no way to query either how many _unclaimed_ rewards remain for a given game, nor the _total number of rewards_ set for it — so there's no way to use the Server API to find out whether one of your games has had _all its rewards claimed._

I can also see this discussion took place a couple of years ago.

Any chance that the idea of a more comprehensive Rewards API is going to be revisited?
